src/Emulator/Cores/RiscV/OpenTitan_PlatformLevelInterruptController.cs

```diff
@@ -0,0 +1,147 @@
+//
+// Copyright (c) 2010-2021 Antmicro
+// Copyright (c) 2021 Google LLC
+//
+// This file is licensed under the MIT License.
+// Full license text is available in 'licenses/MIT.txt'.
+//
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Antmicro.Renode.Core;
+using Antmicro.Renode.Core.Structure.Registers;
+using Antmicro.Renode.Exceptions;
+using Antmicro.Renode.Logging;
+using Antmicro.Renode.Peripherals.Bus;
+using Antmicro.Renode.Utilities;
+using Antmicro.Renode.Peripherals.CPU;
+using Antmicro.Renode.Peripherals.IRQControllers.PLIC;
+
+namespace Antmicro.Renode.Peripherals.IRQControllers
+{
+    [AllowedTranslations(AllowedTranslation.ByteToDoubleWord)]
+    public class OpenTitan_PlatformLevelInterruptController : PlatformLevelInterruptControllerBase, IKnownSize
+    {
+        public OpenTitan_PlatformLevelInterruptController(int numberOfSources = 84, int numberOfContexts = 1)
+            : base(numberOfSources, numberOfContexts, prioritiesEnabled: true,
+                    extraInterrupts: (uint)numberOfContexts)
```

Comment:
Do I understand correctly that OpenTitan PLIC has an additional interrupt for each context - the software interrupt for context?

If so I suggest adding a comment in the code to make it clear. It should also contain a mapping scheme - i.e., which contexts are the "classic" ones and which are the "software interrupt" ones + how do they relate to each other.

Reply:
This is correct.  The CLINT does not really exist in OpenTitan.  In its place are memory mapped registers for each particular hart that are used to interrupt the hart via the MSIP bit in the MIP register.